### PR TITLE
usb: device: nrf5: Fix HF clock enable check

### PR DESCRIPTION
**Pull request by cvinayak**
Fixed return value check after HF clock is requested in
non-blocking mode.

hci_usb sample on nRF52840 SoC failed to enumerate USB
device on cold reset, the clock_control_on interface would
return -EINPROGRESS when clock is enabled in non-blocking
mode.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: drivers/usb/device/usb_dc_nrf5.c [hunk 1]
```diff
@@ -1732,14 +1732,14 @@ int usb_dc_attach(void)
...
-	if (ret) {
+	if (ret && ret != -EINPROGRESS) {
 		goto err_clk_enable;
 	}
 
 	endpoint_ctx_init();
 	ctx->attached = true;
 
-	return ret;
+	return 0;
```
Comment (cvinayak) [inline drivers/usb/device/usb_dc_nrf5.c]: Should -EINPROGRESS be returned here too?

Comment (pawelzadrozniak) [reply drivers/usb/device/usb_dc_nrf5.c]: I think that -EINPROGRESS could be hidden from user here.
The resulting state is the same whether start operation was already in progress or not.

**Review by pawelzadrozniak — APPROVED** [review]

File: drivers/usb/device/usb_dc_nrf5.c (hunk 1 above)
Comment (rakons) [reply drivers/usb/device/usb_dc_nrf5.c]: I can see other problem with this solution - it is laud - I mean, the hf_clock_enable is called, and inside this function errors are printed even if it is ok for the driver to return -EINPROGRESS.

Comment (cvinayak) [reply drivers/usb/device/usb_dc_nrf5.c]: I did not understand, could you ellaborate on the errors?

Comment (rakons) [reply drivers/usb/device/usb_dc_nrf5.c]: The `usb_dc_attach` function calls hf_clock_enable and ignores the error if it gets `-EINPROGRESS` - that part seems ok. But look inside `hf_clock_enable` function:

```c
...
	if (on) {
		ret = clock_control_on(clock, (void *)blocking);
	} else {
		ret = clock_control_off(clock, (void *)blocking);
	}

	if (ret) {
		SYS_LOG_ERR("NRF5 HF clock %s fail: %d",
			    on ? "start" : "stop", ret);
		return ret;
	}
...
```

In this part the error information is generated. So the solution is noisy as you get error message printed even if it is ignored in the caller.

Comment (cvinayak) [reply drivers/usb/device/usb_dc_nrf5.c]: Thanks for catching it... let me see what I can do, meanwhile please put the PR in request for changes.

Comment (pawelzadrozniak) [reply drivers/usb/device/usb_dc_nrf5.c]: Done.

**Review by pawelzadrozniak — CHANGES_REQUESTED** [review]

File: drivers/usb/device/usb_dc_nrf5.c (hunk 1 above)
Comment (cvinayak) [reply drivers/usb/device/usb_dc_nrf5.c]: @pawelzadrozniak  Updated the PR, please review.

**Review by carlescufi — APPROVED** [review]

**Review by pawelzadrozniak — APPROVED** [review]

**Pull request merged**